control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -146,6 +146,11 @@ func ReconcileDeployment(deployment *appsv1.Deployment, ownerRef config.OwnerRef
 			},
 		},
 	}
+	deployment.Spec.Template.Spec.ServiceAccountName = "default"
+	if enableCVOManagementClusterMetricsAccess {
+		deployment.Spec.Template.Spec.ServiceAccountName = manifests.ClusterVersionOperatorServiceAccount("").Name
+		deployment.Spec.Template.Spec.AutomountServiceAccountToken = pointer.Bool(true)
+	}
```

Comment:
@enxebre, the new commit slightly modified already discussed changes from the https://github.com/openshift/hypershift/pull/2807#discussion_r1406039578.

The CVO now has a default `ServiceAccountName = "default"`. An empty  `ServiceAccountName` would not overwrite a non-empty `ServiceAccountName` in case the `EnableCVOManagementClusterMetricsAccess` gets switched off.

The changes were created due to the https://github.com/openshift/hypershift/pull/2807#discussion_r1408559285.